src/libraries/System.IO.Pipes/src/System/IO/Pipes/PipeStream.Unix.cs

```diff
@@ -196,6 +196,12 @@ internal static string GetPipePath(string serverName, string pipeName)
                 throw new ArgumentOutOfRangeException(nameof(pipeName), SR.ArgumentOutOfRange_AnonymousReserved);
             }
 
+            if (RuntimeInformation.RuntimeIdentifier.StartsWith("iossimulator")
+                || RuntimeInformation.RuntimeIdentifier.StartsWith("tvossimulator"))
+            {
+                return $"/tmp/{pipeName}";
```

Comment:
I'm not sure why this is needed?

Reply:
In case of simulators in UnixDomainSocketEndPoint there was ArgumentOutOfRangeException. I added this to make path length shorter because s_nativePathLength is 104 (in UnixDomainSocketEndPoint) and by previous implementation path length is 255.

Reply:
Removed that part. Find out in FileCleanupTestBase path was wrong for osx, updated that.